libs/server/Custom/CustomCommandManagerSession.cs

```diff
@@ -1,100 +1,218 @@
 ﻿// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
-using System.Diagnostics;
+using System;
+using System.Collections.Generic;
 using Garnet.common;
 
 namespace Garnet.server
 {
     /// <summary>
-    /// Server session for RESP protocol - basic commands are in this file
+    /// Session-specific access to custom command data held by CustomCommandManager
+    /// For both custom procedures and custom transactions, this class maintains a cached instance of the transaction / procedure that can be called directly per-session
+    /// This class also maintains a cache of custom command info and docs to avoid the cost of synchronization when calling CustomCommandManager
     /// </summary>
     internal sealed class CustomCommandManagerSession
     {
+        // The instance of the CustomCommandManager that this session is associated with
         readonly CustomCommandManager customCommandManager;
 
-        // These session specific arrays are indexed by the same ID as the arrays in CustomCommandManager
-        ExpandableMap<CustomTransactionProcedureWithArity> sessionTransactionProcMap;
-        ExpandableMap<CustomProcedure> sessionCustomProcMap;
+        // These session specific maps are indexed by the same ID as the arrays in CustomCommandManager
+        // Maps between the transaction ID and a tuple of the per-session transaction procedure and its arity
+        readonly Dictionary<int, (CustomTransactionProcedure, int)> sessionTransactionProcMap;
+        // Maps between the custom procedure ID and the per-session custom procedure instance
+        readonly Dictionary<int, CustomProcedure> sessionCustomProcMap;
+
+        // Cache of custom command info by command name
+        readonly Dictionary<string, RespCommandsInfo> sessionCustomCommandsInfo;
```

Comment:
do we need these two dictionaries? seems like we do not need to access them on critical path as the required info should be within the cmd instance?